src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents the Duke chatbot.
+ */
+public class Duke {
+
+    private static final String FILEPATH = "./data/duke.txt";
+    private static boolean hasNext = true;
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor to initialise a Duke object.
+     *
+     * @param path File path to the text file which stores task information.
+     */
+    public Duke(String path) {
+        this.ui = new Ui();
+        this.storage = new Storage(path);
+
+        try {
+            tasks = new TaskList(storage.loadTaskList());
+        } catch (DukeException e) {
+            ui.printError(e);
```

Comment:
Perhaps can avoid using try-catch as a control flow statement? I believe there is a method to check if the database file exists using Java IO classes 🤔